src/STEMMUS_SCOPE.m

```diff
@@ -46,7 +46,23 @@
 
     % Load model settings: replacing "run Constants"
     ModelSettings = io.getModelSettings();
-    NN = ModelSettings.NN;
+
+    % Load soil layers settings
+    soilLayersFile = [InputPath, 'input_soilLayThick.csv'];
+    if isfile(soilLayersFile)
+        soildata = readmatrix(soilLayersFile);
+        SoilLayerSettings = io.readSoilLayerSettings(soildata);
+        ModelSettings.R_depth = SoilLayerSettings.R_depth;
+        ModelSettings.Tot_Depth = SoilLayerSettings.Tot_Depth;
+        ModelSettings.NL = SoilLayerSettings.NL;
+        ModelSettings.ML = SoilLayerSettings.NL;
+        ModelSettings.DeltZ_R = SoilLayerSettings.DeltZ_R;
+        ModelSettings.DeltZ = flip(SoilLayerSettings.DeltZ_R);
+        ModelSettings.NN = ModelSettings.NL + 1;
+        ModelSettings.mN = ModelSettings.NN + 1;
+        ModelSettings.mL = ModelSettings.NL + 1;
+        ModelSettings.nD = 2;
+    end
```

Comment:
```suggestion
        % Override the default settings
        fields = fieldnames(SoilLayerSettings);
        for i = 1:numel(fields)
            ModelSettings.(fields{i}) = SoilLayerSettings.(fields{i});
        end
    end

    % Calculate other settings
    ModelSettings.NN = ModelSettings.NL + 1;
    ModelSettings.mN = ModelSettings.NN + 1;
    ModelSettings.mL = ModelSettings.NL + 1;
    ModelSettings.nD = 2;
```

Reply:
@MostafaGomaa93 can you please remove [these lines](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/18c39d3ac37a85579c2b5f67c78932ff11733755/src/%2Bio/getModelSettings.m#L77-L80) from `getModelSettings`

Reply:
> @MostafaGomaa93 can you please remove [these lines](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/18c39d3ac37a85579c2b5f67c78932ff11733755/src/%2Bio/getModelSettings.m#L77-L80) from `getModelSettings`

Instead of removing them, we can keep them in the [`getModelSettings`](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/18c39d3ac37a85579c2b5f67c78932ff11733755/src/%2Bio/getModelSettings.m#L77-L80) and add them also to the [`ReadSoilLayerSettings`](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/fbb938b029191da63259829360bcf03d113d0adf/src/%2Bio/readSoilLayerSettings.m#L6C3-L9C30). Then, we don't need to add them in the [`main function`](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/77218e8e498ce26515b532a85a22c130cf7cd9b8/src/STEMMUS_SCOPE.m#L61C2-L65C26)

Reply:
This way you are duplicating code which is not recommended. Please avoid duplicating the code and fix it as suggested.